pkg/rehearse/jobs.go

```diff
@@ -372,39 +374,87 @@ func inlineCiOpConfig(container *v1.Container, ciopConfigs config.DataByFilename
 		logger.WithField(logCiopConfigFile, filename).Debug("Rehearsal job would use ci-operator config from registry, its content will be inlined")
 	}
 
-	ciOpConfigContent, imageStreamTags, err := getResolvedConfigForTest(ciopConfig, resolver, testname)
+	ciOpConfigContent, imageStreamTags, err := getResolvedConfigForTest(ciopConfig, resolver, testName)
 	if err != nil {
 		logger.WithError(err).Error("Failed to get resolved config for test")
 		return nil, err
 	}
 	apihelper.MergeImageStreamTagMaps(allImageStreamTags, imageStreamTags)
+	if !jc.dryRun {
```

Comment:
Upload the config on `!jc.dryRun` makes sense, but I'd expect nothing more than logging when `dryRun == true`. Is perhaps an `enum` more appropriate in this case?

Reply:
The dryRun functionality essentially just means "integration test mode". It is very bad, and that is why I have filed https://issues.redhat.com/browse/DPTP-4243 to completely redesign it.

Reply:
It's misleading but I don't want to delay this PR further just to add one comment. So, it's ok.